### ISSUE TITLE
Redact Sensitive S3 URL Parameters

### ISSUE THREAD

**Issue by bversteegh-xero** [github/gh-gei#1159]
## Description
Raising as a separate ticket on request of @timrogers. This was originally noted as a comment here by @kmaehashi: https://github.com/github/gh-gei/issues/1151#issuecomment-1779001488

On GHES 3.8+, git/metadata archives are uploaded to S3 by GHES, and GHES returns S3 signed URLs for these archives. I think parameters in these URLs (e.g. X-Amz-Credential) should also be masked as well as anyone with access to the GEI log can then access the S3 link
```
[2023-10-25 10:44:25] [INFO] Waiting for archive with id 347 generation to finish. Current status: exported
[2023-10-25 10:44:25] [DEBUG] HTTP GET: https://github-enterprise.example.com/api/v3/orgs/myorgname/migrations/347/archive
[2023-10-25 10:44:25] [DEBUG] GITHUB REQUEST ID: uuid-uuid-uuid
[2023-10-25 10:44:25] [DEBUG] RESPONSE (Redirect): https://example-s3-bucket-name.s3.amazonaws.com/uuid-uuid-uuid.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AAAAAAAAAAAAAAAAAAAAAAA&X-Amz-Date=20231025T104425Z&X-Amz-Expires=172800&X-Amz-Signature=AAAAAAAAAAAAAAAAAAAAAAA&X-Amz-SignedHeaders=host&actor_id=1&key_id=0&repo_id=0&response-content-disposition=filename%3Duuid-uuid-uuid.tar.gz&response-content-type=application%2Fx-gzip
[2023-10-25 10:44:25] [INFO] Archive (metadata) download url: https://example-s3-bucket-name.s3.amazonaws.com/uuid-uuid-uuid.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AAAAAAAAAAAAAAAAAAAAAAA&X-Amz-Date=20231025T104425Z&X-Amz-Expires=172800&X-Amz-Signature=AAAAAAAAAAAAAAAAAAAAAAA&X-Amz-SignedHeaders=host&actor_id=1&key_id=0&repo_id=0&response-content-disposition=filename%3Duuid-uuid-uuid.tar.gz&response-content-type=application%2Fx-gzip
```

**Comment by timrogers** [comment]
Fix in progress in #1169

**Comment by kmaehashi** [comment]
Thank you all for working on this! 👍🏼
